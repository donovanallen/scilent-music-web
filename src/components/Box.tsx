import React from 'react';

import { cn } from '@/lib/utils';

type BoxProps = {
  children: React.ReactNode;
  className?: string;
};

const Box: React.FC<BoxProps> = ({ children, className }) => {
  return (
    <div
      className={cn(
        `rounded-lg h-fit w-full`,
        'bg-light dark:bg-dark',
        className,
      )}
    >
      {children}
    </div>
  );
};

export default Box;
