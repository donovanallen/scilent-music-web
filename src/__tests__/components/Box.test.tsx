import { render } from '@testing-library/react';
import React from 'react';

import Box from '@/components/Box';

describe('Box', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Box>
        <div></div>
      </Box>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
