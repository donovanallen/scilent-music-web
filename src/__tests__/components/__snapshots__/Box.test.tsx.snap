// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
<DocumentFragment>
  <div
    class="bg-dark rounded-lg h-fit w-full"
  >
    <div />
  </div>
</DocumentFragment>
`;
